airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_application_fees.py

```diff
@@ -0,0 +1,376 @@
+        _given_events_availability_check(http_mocker)
+        state_datetime = _NOW - timedelta(days=5)
+        http_mocker.get(
+            _events_request().with_created_gte(state_datetime + _AVOIDING_INCLUSIVE_BOUNDARIES).with_created_lte(_NOW).with_limit(100).with_types(_EVENT_TYPES).build(),
+            _events_response().with_pagination().with_record(
+                _an_event().with_id("last_record_id_from_first_page").with_field(_DATA_FIELD, _an_application_fee().build())
+            ).build(),
+        )
+        http_mocker.get(
+            _events_request().with_starting_after("last_record_id_from_first_page").with_created_gte(state_datetime + _AVOIDING_INCLUSIVE_BOUNDARIES).with_created_lte(_NOW).with_limit(100).with_types(_EVENT_TYPES).build(),
+            _events_response().with_record(
+                self._an_application_fee_event()
+            ).build(),
+        )
+
+        output = self._read(
+            _config(),
+            StateBuilder().with_stream_state(_STREAM_NAME, {"updated": int(state_datetime.timestamp())}).build(),
+        )
+
+        assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_state_and_small_slice_range_when_read_then_perform_multiple_queries(self, http_mocker: HttpMocker) -> None:
+        state_datetime = _NOW - timedelta(days=5)
+        slice_range = timedelta(days=3)
+        slice_datetime = state_datetime + _AVOIDING_INCLUSIVE_BOUNDARIES + slice_range
+
+        _given_application_fees_availability_check(http_mocker)
+        _given_events_availability_check(http_mocker)  # the availability check does not consider the state so we need to define a generic availability check
+        http_mocker.get(
+            _events_request().with_created_gte(state_datetime + _AVOIDING_INCLUSIVE_BOUNDARIES).with_created_lte(slice_datetime).with_limit(100).with_types(_EVENT_TYPES).build(),
+            _events_response().with_record(self._an_application_fee_event()).build(),
+        )
+        http_mocker.get(
+            _events_request().with_created_gte(slice_datetime + _AVOIDING_INCLUSIVE_BOUNDARIES).with_created_lte(_NOW).with_limit(100).with_types(_EVENT_TYPES).build(),
+            _events_response().with_record(self._an_application_fee_event()).with_record(self._an_application_fee_event()).build(),
+        )
+
+        output = self._read(
+            _config().with_start_date(_NOW - timedelta(days=30)).with_slice_range_in_days(slice_range.days),
+            StateBuilder().with_stream_state(_STREAM_NAME, {"updated": int(state_datetime.timestamp())}).build(),
+        )
+
+        assert len(output.records) == 3
+
+    @HttpMocker()
+    def test_given_state_earlier_than_30_days_when_read_then_query_events_using_types_and_event_lower_boundary(self, http_mocker: HttpMocker) -> None:
+        # this seems odd as we would miss some data between start_date and events_lower_boundary. In that case, we should hit the
```

Comment:
is there an issue? I do think the connector should do a full refresh if the input state is too old. this is going to be super important when we start emitting states in full refresh to avoid losing data if a sync transiently fails

A potential optimization would be to do a full refresh + client side filtering, but that seems less important than the not losing data part

Reply:
This is indeed a potential issue today (if the sync is less frequent than 30 days). It will be even worst of an issue for the case you mentioned. Ticket created: https://github.com/airbytehq/airbyte/issues/33678